device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsAuthenticationLinkHandlerX509.java

```diff
@@ -7,31 +7,23 @@
 import org.apache.qpid.proton.engine.SslDomain;
 import org.apache.qpid.proton.engine.Transport;
 
-import java.io.IOException;
+import javax.net.ssl.SSLContext;
 import java.util.UUID;
 
 @Slf4j
-public class AmqpsDeviceAuthenticationX509 extends AmqpsDeviceAuthentication
+public class AmqpsAuthenticationLinkHandlerX509 extends AmqpsAuthenticationLinkHandler
```

Comment:
What does this link do? I thought X509 certificate info is handled by the tcp connection directly (from the ssl context supplied)

Reply:
This class doesn't do much, but it does differ from the CBS implementation of the AmqpsAuthenticationLinkHandler in how it sets the SSL domain. For CBS, during the SSL domain setup, you enable SASL auth with type "anonymous". X509 doesn't do that. Really, its a bit of an odd class overall since x509 auth has no links to manage. What do you think about it?